### PR TITLE
One capability failure shoud not affect the next

### PR DESCRIPTION
**Pull request by greeshmaswaminathan**
Cherry pick for https://github.com/cdapio/cdap/pull/13049

### CODE REVIEW

**Review by pandyajay10 — APPROVED** [review]

File: cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplier.java [hunk 1]
```diff
@@ -311,8 +310,7 @@ private void doForAllAppsWithCapability(String capability, CheckedConsumer<Appli
 
   private boolean shouldRetry(Throwable throwable) {
     return !(throwable instanceof UnauthorizedException ||
-      throwable instanceof InvalidArtifactException ||
-      throwable instanceof ArtifactNotFoundException);
```
Comment (yaojiefeng) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplier.java:311]: What if the artifact is deleted and gone? Should we retry forever?

Comment (yaojiefeng) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplier.java:311]: Seems like there is a retry limit, please ignore this comment

Comment (greeshmaswaminathan) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/CapabilityApplier.java:311]: There is a retry limit which is set currently at 5.

**Pull request merged**